packages/ui-components/src/Input/NumberInput.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useState } from "react";
+import classNames from "classnames/bind";
+import { CaretUp, CaretDown } from "@cockroachlabs/icons";
+import isNumber from "../utils/isNumber";
+import { BaseInput, InputProps } from "./BaseInput";
+import styles from "./styles.module.scss";
+import { InputPrefix, InputWrapper } from "./helpers";
+
+const cx = classNames.bind(styles);
+
+export type NumberInputProps = Omit<InputProps<number>, "type">;
+
+export const NumberInput: React.FC<NumberInputProps> = ({
+  onChange,
+  value: outerValue,
+  initialValue,
+  prefix,
+  ...props
+}) => {
+  const { invalid, disabled } = props;
```

Comment:
Why  destructure  these here and not up in the parameter list with the other?

Reply:
Good point! The reason was to pass rested `props` to `<BaseInput {...props} ... />` and have these props inside rested object. But it already has other props defined, so will destruct it in the parameters list. Thanks!